csrc/cutlass/block_scaling.cpp

```diff
@@ -0,0 +1,326 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <cutlass/block_scaling.h>
+#include <exceptions.h>
+#include <fusion.h>
+#include <ir/interface_nodes.h>
+#include <ir/internal_nodes.h>
+#include <ir/utils.h>
+#include <type.h>
+
+namespace nvfuser {
+
+namespace {
+
+// These are the quantized types that make use of block scaling for which we
+// have established Fusion IR recipes.
+bool isBlockScaledDtype(const DataType& dtype) {
+  return dtype == DataType::Float4_e2m1fn || dtype == DataType::Float8_e4m3fn ||
+      dtype == DataType::Float8_e5m2;
+}
+
+} // namespace
+
+namespace cutlass_codegen {
+
+std::string BlockScaledOutputPattern::toString() const {
+  std::stringstream ss;
+  ss << "BlockScaledOutputPattern: {\n";
+  ss << "  unquantized_output: " << unquantized_output->toString() << "\n";
+  ss << "  quantized_output: " << quantized_output->toString() << "\n";
+  ss << "  block_scale_factors: " << block_scale_factors->toString() << "\n";
+  ss << "  block_size: " << block_size << "\n";
+  ss << "}\n";
+  return ss.str();
+}
+
+// This matches a standard block scaling pattern expressed in Fusion IR.
+//
+// Block scaling quantizes a tensor by dividing it by per-block scale factors
+// that are computed from the absolute maximum values within each block.
+//
+// Pattern diagram (without global scale):
+//
+//          unquantized_output (high precision)
```

Comment:
```suggestion
#if defined(__clang__) || defined(__GNUC__)
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wcomment"
#endif
```
**[Suggested by AI]** Summary of change:
• Added compiler-specific pragmas that are only enabled when the code is compiled with Clang or GCC.  
• The new block pushes the current diagnostic state and then disables the “-Wcomment” warning:  

  #if defined(__clang__) || defined(__GNUC__)
  #pragma GCC diagnostic push
  #pragma GCC diagnostic ignored "-Wcomment"
  #endif

Purpose/impact:
• Suppresses the “-Wcomment” warning (e.g., warnings about nested or malformed comments) that was triggering a build failure with GCC/Clang.  
• Uses diagnostic push so the warning state can later be restored (presumably with a corresponding #pragma GCC diagnostic pop), limiting the suppression to the specific code section.  
• Has no effect when compiled with other compilers, so it fixes the build error without affecting non-GCC/Clang toolchains.